web/src/lib/utils/thumbnail-util.ts

```diff
@@ -43,33 +43,44 @@ export const getAltText = derived(t, ($t) => {
       return asset.exifInfo.description;
     }
 
-    let altText = $t('image_taken', { values: { isVideo: asset.type === AssetTypeEnum.Video } });
+    const date = fromLocalDateTime(asset.localDateTime).toLocaleString({ dateStyle: 'long' });
+    const hasLocation = asset.exifInfo?.city && asset.exifInfo?.country;
+    const names = asset.people?.filter((p) => p.name).map((p) => p.name) ?? [];
+    const peopleCount = names.length;
+    const isVideo = asset.type === AssetTypeEnum.Video;
 
-    if (asset.exifInfo?.city && asset.exifInfo?.country) {
-      const placeText = $t('image_alt_text_place', {
-        values: { city: asset.exifInfo.city, country: asset.exifInfo.country },
-      });
-      altText += ` ${placeText}`;
-    }
+    let key = 'image_alt_text_date';
+    const values = {
+      date,
+      city: asset.exifInfo?.city,
+      country: asset.exifInfo?.country,
+      person1: names[0],
+      person2: names[1],
+      person3: names[2],
+      isVideo,
+      additionalCount: peopleCount > 3 ? peopleCount - 2 : 0,
+    };
 
-    const names = asset.people?.filter((p) => p.name).map((p) => p.name) ?? [];
-    if (names.length > 0) {
-      const namesText = $t('image_alt_text_people', {
-        values: {
-          count: names.length,
-          person1: names[0],
-          person2: names[1],
-          person3: names[2],
-          others: names.length > 3 ? names.length - 2 : 0,
-        },
-      });
-      altText += ` ${namesText}`;
+    if (hasLocation) {
+      key += '_place';
     }
 
-    const date = fromLocalDateTime(asset.localDateTime).toLocaleString({ dateStyle: 'long' });
-    const dateText = $t('image_alt_text_date', { values: { date } });
-    altText += ` ${dateText}`;
+    if (peopleCount > 0) {
+      key += getPersonSuffix(peopleCount);
+    }
 
-    return altText;
+    return $t(key, { values });
   };
 });
+
+const getPersonSuffix = (count: number) => {
+  if (count === 1) {
+    return '_1_person';
+  }
+
+  if (count === 2 || count === 3) {
+    return `_${count}_people`;
+  }
+
+  return '_4_or_more_people';
+};
```

Comment:
I am not a fan of dynamically computed translation keys. It makes it really hard to find in the code and for other tools to integrate with.

Reply:
Yeah I search by translation keys all the time, thanks for the feedback! I'll get this fixed.